src/components/FeatureViews/Dashboard/Dashboard.js

```diff
@@ -38,21 +39,59 @@ const useStyles = makeStyles((theme) => ({
 
 const Dashboard = () => {
     const classes = useStyles();
-    const user = useSelector(({auth}) => auth) || [];
+    const {email} = useSelector(({auth}) => auth) || [];
+    const emailQuery = { "email": email }
+    const USER_QUERY = gql`query userQuery($email: String="") {
```

Comment:
You shouldn't need this query, just add more to the GET_ACCOUNT_TYPE query in authActions.js or just change authActions.js after line 5 to this - this includes a change to query for more information. With this extra information, you can use line 42 for more than just the email
```
const GET_EMAIL = gql`
    mutation GetDetails($token: String!) {
        verifyToken(token: $token) {
            payload
        }
    }`;

const GET_ACCOUNT_TYPE = gql`
    query GetAccountType($username: String!) {
        userInfo(userName: $username) {
            ... on AdminType {
                accountType
                user {
                    id
                    firstName
                    lastName
                    email
                }
            }
            ... on InstructorType {
                accountType
                phoneNumber
                user {
                    id
                    firstName
                    lastName
                    email
                }
            }
            ... on ParentType {
                accountType
                phoneNumber
                user {
                    id
                    firstName
                    lastName
                    email
                }
            }
        }
    }`;

export const setToken = async (token, shouldSave) => {
    try {
        const {"data": {verifyToken}} = await client.mutate({
            "context": {
                "headers": {
                    "Authorization": `JWT ${token}`,
                },
            },
            "mutation": GET_EMAIL,
            "variables": {token},
        });

        const email = verifyToken.payload.username;

        const {"data": {userInfo}} = await client.query({
            "context": {
                "headers": {
                    "Authorization": `JWT ${token}`,
                },
            },
            "query": GET_ACCOUNT_TYPE,
            "variables": {"username": email},
        });
		const {accountType, user, phoneNumber} = userInfo;

        if (shouldSave) {
            localStorage.setItem("token", token);
        }
        return {
            "payload": {
                accountType,
                email,
                token,
				user,
                phoneNumber
            },
            "type": types.SET_CREDENTIALS,
        };
    } catch (error) {
        // invalid token, do nothing
        console.error(error);
        return {
            "type": null,
        };
    }
};

export const logout = () => {
    localStorage.removeItem("token");
    client.clearStore();
    return {
        "type": types.LOGOUT,
    };
};
```